concepts/05 UI Components/DataGrid/20 Editing/43 Customize Edit Form/30 Save, Cancel, and Custom Buttons.md

```diff
@@ -0,0 +1,23 @@
+Form and popup editing modes include built-in Save and Cancel buttons. If you want to execute custom code on click one of these buttons, specify the [onSaving](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onSaving) or [onEditCanceling](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditCanceling) event handler correspondingly. Assign `true` to the **e.cancel** parameter to cancel the Save or Cancel operations and keep the form or popup opened.
```

Comment:
```suggestion
Form and popup edit modes include built-in Save and Cancel buttons. If you want to execute custom code on a click on either button, specify the [onSaving](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onSaving) or [onEditCanceling](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditCanceling) event handler, respectively. Assign `true` to the **e.cancel** parameter to cancel a pending Save or Cancel operation and keep the form or popup window open.
```

P.S. If you write "popup" without a hyphen throughout your documentation repository every time you use this word as an adjective -- then everything is fine. Otherwise, you may want to replace "popup" with "pop-up". Just checking.